src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -229,6 +230,21 @@ def app_teardown(
 @app.command("deploy", requires_connection=True)
 @with_project_definition("native_app")
 def app_deploy(
+    prune: Optional[bool] = typer.Option(
+        default=None,
+        help=f"""Controls whether files that exist only remotely will be deleted from the stage. When specific files are specified, the default is --no-prune. When no files are specified (sync all changes), the default is --prune.""",
+    ),
+    recursive: Optional[bool] = typer.Option(
+        None,
+        "--recursive",
+        "-r",
+        help=f"""Controls whether the specified directories should be deployed including all of their contents. When no files are specified (sync all changes), the default is --recursive.""",
+    ),
+    files: Optional[List[Path]] = typer.Argument(
+        default=None,
+        show_default=False,
+        help=f"""Paths of the files, relative to the project root, to be uploaded to a stage. [default: sync all local changes to the stage]""",
+    ),
```

Comment:
Wouldn't it be easier to understand if we say that --prune always default to false, --recursive always defaults to false, but deploy with no args is a shorthand for --prune --recursive? This way, there's only one special case: absolutely no arguments (so app deploy --prune implies --no-recursive, is what I'm saying, because an argument was specified and --no-recursive is always the default). Otherwise, you have to worry about files/no-files, which I think is a lot more confusing.

Reply:
Changed it in [this commit](https://github.com/snowflakedb/snowflake-cli/pull/933/commits/8d89da9ef210cf22cc5fe50c48d9af474792912f). WDYT? I'm not sure it's that much simpler, but I'm ok with both ways.